app/src/main/java/com/practicum/playlistmaker/SearchActivity.kt

```diff
@@ -29,41 +36,139 @@ class SearchActivity : AppCompatActivity() {
 
         val inputEditText = binding.searchEtInputSeacrh
         val searchClearButton = binding.searchIvClearIcon
+        val placeholderUpdateButton = binding.searchBvPlaceholderButton
+
+        if (searchInput.isNotEmpty()) {
+            inputEditText.setText(searchInput)
+        }
 
         searchClearButton.setOnClickListener {
             inputEditText.setText("")
             hideKeyboard()
+            trackList.clear()
+            adapter.notifyDataSetChanged()
+            placeholderVisibility(false, false, false, false)
         }
 
         binding.searchToolbar.setNavigationOnClickListener() {
             finish()
         }
-        if (searchInput.isNotEmpty()) {
-            inputEditText.setText(searchInput)
+
+        placeholderUpdateButton.setOnClickListener {
+            searchInITunes(inputEditText.text.toString())
+        }
+
+
+        binding.searchEtInputSeacrh.setOnEditorActionListener { _, actionId, _ ->
+            if (actionId == EditorInfo.IME_ACTION_DONE) {
+                if (inputEditText.text.isNotEmpty()) {
+                    searchInITunes(inputEditText.text.toString())
+                }
+            }
+            false
         }
 
         val searchTextWatcher = object : TextWatcher {
             override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
-                // empty
             }
 
             override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
-
                 searchClearButton.isVisible = !s.isNullOrEmpty()
             }
 
+            @SuppressLint("NotifyDataSetChanged")
             override fun afterTextChanged(s: Editable?) {
                 searchInput = s.toString()
+                if (searchInput.isNotEmpty()) {
+                    searchInITunes(inputEditText.text.toString())
+
+                } else {
+                    trackList.clear()
+                    adapter.notifyDataSetChanged()
+                }
             }
         }
 
         inputEditText.addTextChangedListener(searchTextWatcher)
 
 
-        val recyclerView = binding.searchRcSearchResults
-        recyclerView.layoutManager = LinearLayoutManager(this)
-        recyclerView.adapter = SearchResultsAdapter()
+        val tracksRecView = binding.searchRcSearchResults
+        tracksRecView.layoutManager = LinearLayoutManager(this)
+        tracksRecView.adapter = adapter
+
+
+    }
+
+    fun searchInITunes(text: String) {
+        iTunesService.search(text)
+            .enqueue(object : Callback<SongsResponse> {
+                @SuppressLint("NotifyDataSetChanged")
+                override fun onResponse(
+                    call: Call<SongsResponse>,
+                    response: Response<SongsResponse>
+                ) {
+                    if (response.code() == 200) {
```

Comment:
```suggestion
                    if (response.isSuccessfull) {
```
Чтобы не хардкодить в активити коды ответов http запросов